amads/complexity/mtype_features.py

```
@@ -0,0 +1,386 @@
+    - :math:`m` is the index for the frequency class in the frequency distribution
+
+    Parameters
+    ----------
+    ngram_counts : dict[tuple, int]
+        Dictionary of n-gram counts
+
+    Returns
+    -------
+    float
+        Mean Yule's K statistic across all n-gram lengths.
+        Returns 0 for empty input.
+    Examples
+    --------
+    >>> from amads.algorithms.mtype_tokenizer import FantasticTokenizer
+    >>> from amads.core.basics import Score
+    >>> # Sample melody [60, 62, 64, 66, 64, 62, 60] with equal note durations
+    >>> score = Score.from_melody([60, 62, 64, 66, 64, 62, 60], durations=1.0)
+    >>> tokenizer = FantasticTokenizer()
+    >>> high_repetition_counts = tokenizer.get_mtype_counts(score, method="all")
+    >>> yules_k(high_repetition_counts)  # High K indicates more repetition
+    22.68...
+    >>> # Sample melody with low repetition [60, 61, 62, 68, 59, 71] with different durations
+    >>> score = Score.from_melody([60, 61, 62, 68, 59, 71], durations=[0.25, 0.5, 0.75, 0.35, 0.15, 0.25])
+    >>> tokenizer = FantasticTokenizer()
+    >>> low_repetition_counts = tokenizer.get_mtype_counts(score, method="all")
+    >>> yules_k(low_repetition_counts)  # Low K indicates less repetition
+    8.88...
+    >>> # Empty input
+    >>> yules_k({})
+    0.0
+    """
+    if not ngram_counts:
+        return 0.0
+
+    k_values = []
+
+    # Get frequency of frequencies
+    freq_spec = Counter(ngram_counts.values())
+
+    # Calculate total_tokens (total tokens)
+    total_tokens = sum(ngram_counts.values())
+    if total_tokens == 0:
+        return 0.0
+
+    # Calculate sum(vm * m²) where vm is frequency of value m
+    vm_m2_sum = sum(freq * (count * count) for count, freq in freq_spec.items())
+
+    # Calculate K with scaling factor of 1000
+    k = 1000 * (vm_m2_sum - total_tokens) / (total_tokens * total_tokens)
+    k_values.append(k)
+
+    return float(np.mean(k_values)) if k_values else 0.0
```

I'm not sure about silently returning 0.0 in an empty sequence -- maybe raise a ValueError instead?

---

Yeah, I think you're right, we want people to know that the calculation has failed, rather than thinking that 0.0 is a real calculated value